src/config.rs

```diff
@@ -118,19 +119,26 @@ impl SerialisableCertificate {
     /// Returns [CertificateParseError](enum.Error.html#variant.CertificateParseError) if the inputs
     /// cannot be parsed
     pub fn obtain_priv_key_and_cert(&self) -> R<(quinn::PrivateKey, quinn::Certificate)> {
+        // FIXME: Seems quinn have `ParseError` as a return type of a public interface but the type
+        // itself is private. Hence using this workaround to collect it via `format!`. Ideally should
+        // be just convertible inside quick_error as usual with other errors.
         Ok((
-            quinn::PrivateKey::from_der(&self.key_der)?,
-            quinn::Certificate::from_der(&self.cert_der)?,
+            quinn::PrivateKey::from_der(&self.key_der)
+                .map_err(|e| Error::CertificateParseError(format!("{:?}, {}", e, e)))?,
+            quinn::Certificate::from_der(&self.cert_der)
+                .map_err(|e| Error::CertificateParseError(format!("{:?}, {}", e, e)))?,
         ))
     }
 }
 
 impl Default for SerialisableCertificate {
     fn default() -> Self {
-        let cert = rcgen::generate_simple_self_signed(vec!["MaidSAFE.net".to_string()]);
+        let cert = unwrap!(rcgen::generate_simple_self_signed(vec![
```

Comment:
We need to handle this error.

Reply:
Might need a slight code rewrite here because this one is in the `Default` impl which doesn't return a `Result`, so I'd prefer to tackle it in a separate PR along with quick_error removal if you don't mind?